edvart/report_sections/univariate_analysis.py

```diff
@@ -384,29 +386,37 @@ def add_cells(self, cells: List[Dict[str, Any]]) -> None:
                 column_header = nbfv4.new_markdown_cell(f"## *{col} - {data_type_name}*")
                 cells.append(column_header)
                 if data_type in (DataType.CATEGORICAL, DataType.BOOLEAN):
-                    code = code_dedent(
-                        f"""
-                        top_most_frequent(df['{col}'])
-                        bar_plot(df['{col}'])"""
-                    )
-                elif self.verbosity == 1:
-                    code = code_dedent(
-                        f"""
-                            numeric_statistics(df['{col}'])
-                            histogram(df['{col}'])"""
+                    cell = nbfv4.new_code_cell(
+                        code_dedent(
+                            f"""
+                            top_most_frequent(df['{col}'])
+                            bar_plot(df['{col}'])"""
+                        )
                     )
+                elif data_type == DataType.UNIQUE:
+                    cell = nbfv4.new_markdown_cell("Each value in the column is unique.")
                 else:
-                    code = code_dedent(
-                        f"""
-                            numeric_statistics(
-                                df['{col}'],
-                                descriptive_stats=default_descriptive_statistics(),
-                                quantile_stats=default_quantile_statistics()
+                    if self.verbosity == 1:
+                        cell = nbfv4.new_code_cell(
+                            code_dedent(
+                                f"""
+                                numeric_statistics(df['{col}'])
+                                histogram(df['{col}'])"""
                             )
-                            histogram(df['{col}'])"""
-                    )
-                code_cell = nbfv4.new_code_cell(code)
-                cells.append(code_cell)
+                        )
+                    else:
+                        cell = nbfv4.new_code_cell(
+                            code_dedent(
+                                f"""
+                                numeric_statistics(
+                                    df['{col}'],
+                                    descriptive_stats=default_descriptive_statistics(),
+                                    quantile_stats=default_quantile_statistics()
+                                )
+                                histogram(df['{col}'])"""
+                            )
+                        )
+                cells.append(cell)
```

Comment:
Just checking: is changing `elif self.verbosity == 1:` into the code below just a refactor or does it change the functionality? I am not sure if I am not missing something.
```python
else:
     if self.verbosity == 1:
        ....
     else
         ....
```
If it's just a refactor I would do it. `elif` is IMHO better.

Reply:
The functionality is the same as before. I split it to separate two kinds of conditions mixed in the quite long `if-elif` chain:
- First there are conditions according to the data type
- If it's not any of the data types checked then the verbosity is checked.
I found it a bit confusing when it was one long chain so I split it like this.